### ISSUE TITLE
`vue/no-unused-properties` does not work with `defineNuxtComponent()`

### ISSUE THREAD

**Issue by uralogical** [vuejs/eslint-plugin-vue#2191]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:**  `8.41.0`
- **eslint-plugin-vue version:** `9.14.0`
- **Node version:** `18`
- **Operating System:** MacOS 13.3.1 (a)（22E772610a）

**Please show your full configuration:**

```json5
module.exports = {
  extends: [
    "plugin:vue/vue3-recommended",
  ],
  rules: {
    "vue/no-unused-properties": [
      "error",
      {
        groups: ["props", "data", "computed", "methods"],
        deepData: false,
        ignorePublicMembers: false,
      },
    ],
  },
};
```

**What did you do?**	

```vue
<template>
  <div>
    <span>Hello</span>
  </div>
</template>

<script lang="ts">
export default defineNuxtComponent({
  name: 'App',
  data() {
    return {
      msg: 'Welcome to Nuxt3', // This data is not used.
    };
  },
});
</script>
```

**What did you expect to happen?**
Data `msg` should be warned like below.
This warning is shown without `defineNuxtComponent`.

<img width="496" alt="スクリーンショット 2023-05-25 21 34 25" src="https://github.com/vuejs/eslint-plugin-vue/assets/12651573/fd315653-e101-4b13-8f1c-d3bd0acfe381">

It seems to be rerated to https://github.com/vuejs/eslint-plugin-vue/issues/2057.

**What actually happened?**
No warning is shown.

**Repository to reproduce this issue**
https://github.com/uralogical/nuxt3_playground
`feat/add_eslint` is reproduction branch.

**Comment by FloEdelmann** [comment]
`defineNuxtComponent` is not yet supported. We might add this for v10, see #2166.

**Comment by IlyaSemenov** [comment]
FWIW, I'm using this for now:

```ts
import { defineNuxtComponent as defineComponent } from '#imports'

export default defineComponent({
  ...
})
```

**Comment by sugitata** [comment]
my workaround https://github.com/vuejs/eslint-plugin-vue/pull/2311

**Comment by FloEdelmann** [comment]
Will be fixed in v10.
